wallet/reference/non-evm-apis/starknet-snap-api.md

```diff
@@ -0,0 +1,867 @@
+      },
+    },
+  },
+})
+```
+
+</TabItem>
+<TabItem value="Result">
+
+```json
+{
+  "transaction_hash": "0x05a56e2d52c817161883f50c441c3228cfe54d9f84b5b5b8b1c8b8e0e6f7e6d8",
+  "address": "0xb60e8dd61c5d32be8058bb8eb970870f07233155"
+}
+```
+
+</TabItem>
+</Tabs>
+
+### `starkNet_displayPrivateKey`
+
+Extracts the private key from the deployed Starknet account and displays it in MetaMask.
+
+#### Parameters
+
+- `userAddress`: `string` - Address of the account contract.
+- `chainId`: `string` - (Optional) ID of the target Starknet network.
+  The default is the Starknet Sepolia testnet.
+
+#### Returns
+
+Always returns `null` for security reasons.
+The private key is only shown in the MetaMask pop-up window.
+
+#### Example
+
+<Tabs>
+<TabItem value="Request">
+
+```js
+await getEip6963Provider.request({
+  method: "wallet_invokeSnap",
+  params: {
+    snapId: "npm:@consensys/starknet-snap",
+    request: {
+      method: "starkNet_displayPrivateKey",
+      params: {
+        userAddress: "0xb60e8dd61c5d32be8058bb8eb970870f07233155",
+        "chainId": "0x534e5f5345504f4c4941"
+      },
+    },
+  },
+})
+```
+
+</TabItem>
+<TabItem value="Result">
+
+```json
+null
+```
+
+</TabItem>
+</Tabs>
+
+### `starkNet_estimateAccountDeployFee`
+
+Gets the estimated gas fee for deploying an account contract.
+
+#### Parameters
+
+- `addressIndex`: `integer` - (Optional) Specific address index of the derived key in
+  [BIP-44](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki).
+- `chainId`: `string` - (Optional) ID of the target Starknet network.
+  The default is the Starknet Sepolia testnet.
+
+#### Returns
+
+The response from Starknet's `feeder_gateway/estimate_fee` API endpoint.
+
+#### Example
+
+<Tabs>
+<TabItem value="Request">
+
+```js
+await getEip6963Provider.request({
+  method: "wallet_invokeSnap",
+  params: {
+    snapId: "npm:@consensys/starknet-snap",
+    request: {
+      method: "starkNet_estimateAccountDeployFee",
+      params: {
+        addressIndex: 0,
+        "chainId": "0x534e5f5345504f4c4941"
+      },
+    },
+  },
+})
+```
+
+</TabItem>
+<TabItem value="Result">
+
+```json
+{
+  "suggestedMaxFee": "1000000000000000",
+  "overallFee": "900000000000000",
+  "gasConsumed": "1000000",
+  "gasPrice": "1000000000",
+  "unit": "wei"
+}
+```
+
+</TabItem>
+</Tabs>
+
+### `starkNet_estimateFee`
+
+Gets the estimated gas fee for calling a method on any contract.
+
+#### Parameters
+
+- `contractAddress`: `string` - Address of the target contract.
+- `contractFuncName`: `string` - Target function name of the contract.
+- `contractCallData`: `string` - (Optional) Call data for the target function with `,` as a separator.
+- `senderAddress`: `string` - Address of the sender.
+- `chainId`: `string` - (Optional) ID of the target Starknet network.
+  The default is the Starknet Sepolia testnet.
+
+#### Returns
+
+The response from Starknet's `feeder_gateway/estimate_fee` API endpoint.
```

Comment:
Since the `feeder_gateway`/`gateway` has been [deprecated](https://community.starknet.io/t/feeder-gateway-deprecation/100233), should this be updated?